broker-core/src/main/java/io/zeebe/broker/logstreams/processor/TypedCommandWriterImpl.java

```diff
@@ -105,7 +109,7 @@ protected void appendRecord(
       event.key(key);
     } else {
       if (type == RecordType.EVENT) {
-        event.positionAsKey();
+        event.key(keyGenerator.nextKey());
```

Comment:
Do we still want to generate keys implicitly in the writer? Since we have to pass a key for new events in `TypedStreamWriter`, the key should be generated before.

Reply:
You should always generate the key before, I just thought that given issue #2016 the fallback should be a newly generated key. It can be -1 but that also isn't unique.